src/AssetsController.ts

```diff
@@ -0,0 +1,265 @@
+	 * @returns - Current collectible list
+	 */
+	async addCollectible(address: string, tokenId: number): Promise<Collectible[]> {
+		address = toChecksumAddress(address);
+		const { allCollectibles, collectibles } = this.state;
+		const { networkType, selectedAddress } = this.config;
+		const existingEntry = collectibles.find(
+			(collectible) => collectible.address === address && collectible.tokenId === tokenId
+		);
+		if (existingEntry) {
+			return collectibles;
+		}
+		const { name, image } = await this.requestNFTCustomInformation(address, tokenId);
+		const newEntry: Collectible = { address, tokenId, name, image };
+		const newCollectibles = [...collectibles, newEntry];
+		const addressCollectibles = allCollectibles[selectedAddress];
+		const newAddressCollectibles = { ...addressCollectibles, ...{ [networkType]: newCollectibles } };
+		const newAllCollectibles = { ...allCollectibles, ...{ [selectedAddress]: newAddressCollectibles } };
+		this.update({ allCollectibles: newAllCollectibles, collectibles: newCollectibles });
+		return newCollectibles;
+	}
+
+	/**
+	 * Removes a token from the stored token list
+	 *
+	 * @param address - Hex address of the token contract
+	 */
+	removeToken(address: string) {
+		address = toChecksumAddress(address);
+		const { allTokens, tokens } = this.state;
+		const { networkType, selectedAddress } = this.config;
+		const newTokens = tokens.filter((token) => token.address !== address);
+		const addressTokens = allTokens[selectedAddress];
+		const newAddressTokens = { ...addressTokens, ...{ [networkType]: newTokens } };
+		const newAllTokens = { ...allTokens, ...{ [selectedAddress]: newAddressTokens } };
+		this.update({ allTokens: newAllTokens, tokens: newTokens });
+	}
+
+	/**
+	 * Removes a collectible from the stored token list
+	 *
+	 * @param address - Hex address of the collectible contract
+	 * @param tokenId - Token identifier of the collectible
+	 */
+	removeCollectible(address: string, tokenId: number) {
+		address = toChecksumAddress(address);
+		const { allCollectibles, collectibles } = this.state;
+		const { networkType, selectedAddress } = this.config;
+		const newCollectibles = collectibles.filter(
+			(collectible) => !(collectible.address === address && collectible.tokenId === tokenId)
+		);
+		const addressCollectibles = allCollectibles[selectedAddress];
+		const newAddressCollectibles = { ...addressCollectibles, ...{ [networkType]: newCollectibles } };
+		const newAllCollectibles = { ...allCollectibles, ...{ [selectedAddress]: newAddressCollectibles } };
+		this.update({ allCollectibles: newAllCollectibles, collectibles: newCollectibles });
+	}
+
+	/**
+	 * Request NFT custom information of a collectible
+	 *
+	 * @param address - Hex address of the collectible contract
+	 * @param tokenId - The NFT identifier
+	 * @returns - Current collectible name and image
+	 */
+	async requestNFTCustomInformation(address: string, tokenId: number): Promise<CollectibleCustomInformation> {
```

Comment:
Do you think this method should be public? Will it ever be called externally? Same question for `fetchCollectibleBasicInformation`. If these methods will only ever be used internally to this controller, I say we make them private to keep the external API as small as possible.

Reply:
totally agree, I'll move it